GoogleForCreators/web-stories-wp
# Document Panel: Update Page Advancement Panel

## Context

- Update the page advancement panel to match the [new designs](https://www.figma.com/file/bMhG3KyrJF8vIAODgmbeqT/Design-System?node-id=3398%3A94450)

![image](https://user-images.githubusercontent.com/22185279/111659411-d452a500-87d2-11eb-96ef-8bc6a2767158.png)

## Summary

- Update page advancement panel
    - Replace old `Switch` with new `Switch` in the editor
    - Remove the `RangeInput`
    - Replace `BoxedNumeric` with `NumericInput`
- Update new `Switch` component to better match the new [switch designs](https://www.figma.com/file/bMhG3KyrJF8vIAODgmbeqT/Design-System?node-id=3447%3A64541)

## Relevant Technical Choices

Moved the `useKeyDownEffect` from the switch in the design system to the switch wrapper in the editor. The editor wants the extra functionality, but I think it makes more sense to keep the base `Switch` as composable as possible.

## To-do

Switch designs are still in flux (hover, disabled states). This pr doesn't depend on those, but they will need to be added

## User-facing changes

|Before|After|
|--|--|
|![page-advancement-old](https://user-images.githubusercontent.com/22185279/111660309-91dd9800-87d3-11eb-8053-ddd1d4e00b0a.gif)|![page-advancement-new](https://user-images.githubusercontent.com/22185279/111660323-943ff200-87d3-11eb-877b-68f676942873.gif)|
|![page-advancement-rtl-old](https://user-images.githubusercontent.com/22185279/111660910-2b0cae80-87d4-11eb-94fd-d24535a690ff.gif)|![page-advancement-rtl](https://user-images.githubusercontent.com/22185279/111660915-2d6f0880-87d4-11eb-8868-7b4108249ac8.gif)|



## Testing Instructions

This PR can be tested by following these steps:

1. Create a new story/open existing story
2. Click the `Document` tab to open the document panel
3. Scroll to bottom
4. Play with the `Page Advancement` panel. Should be keyboard accessible
    - up/down/left/right change the slider value
    - space + enter change the slider value (extra functionality)


### QA






This PR can be tested by following these steps:

1. Create a new story/open existing story
2. Click the `Document` tab to open the document panel
3. Scroll to bottom
4. Play with the `Page Advancement` panel

### UAT






## Reviews

### Does this PR have a security-related impact?

no

### Does this PR change what data or activity we track or use?

no

### Does this PR have a legal-related impact?

no

## Checklist




---



Fixes #6519
Fixes #6492
